src/components/buy/giro-code.tsx

```diff
@@ -6,39 +6,13 @@ interface GiroCodeProps {
 }
 
 export function GiroCode({ info }: GiroCodeProps): JSX.Element {
-  const defaultValues = {
-    service: 'BCD',
-    version: '001',
-    encoding: '2',
-    transfer: 'SCT',
-    char: '',
-    ref: '',
-  };
-
   function isValid(): boolean {
-    return Boolean(info.currency) && Boolean(info.amount);
-  }
-
-  function toValue(info: PaymentInformation): string {
-    if (!isValid()) return '';
-    return `
-    ${defaultValues.service}
-    ${defaultValues.version}
-    ${defaultValues.encoding}
-    ${defaultValues.transfer}
-    ${info.bic}
-    ${info.recipient}
-    ${info.iban}
-    ${info.currency?.name}${info.amount}
-    ${defaultValues.char}
-    ${defaultValues.ref}
-    ${info.purpose}
-    `.trim();
+    return Boolean(info.giroCode);
   }
 
   return isValid() ? (
     <div className="flex flex-col items-center py-4 gap-1">
-      <QRCode className="mx-auto" value={toValue(info)} size={128} fgColor={'#072440'} />
+      <QRCode className="mx-auto" value={info.giroCode ?? ''} size={128} fgColor={'#072440'} />
       <p className="text-dfxBlue-800 font-semibold text-base">GiroCode</p>
     </div>
```

Comment:
If you replace `isValid()` by `info.giroCode != null`, you don't have to do `info.giroCode ?? ''` 😉